source/mqtt/MqttClient.cpp

```diff
@@ -694,6 +694,15 @@ namespace Aws
                 const Crt::Io::TlsContext &tlsContext,
                 bool useWebsocket) noexcept
             {
+                if (!tlsContext)
+                {
+                    AWS_LOGF_ERROR(
+                        AWS_LS_MQTT_CLIENT,
+                        "id=%p Trying to call MqttClient::NewConnection using an invalid TlsContext.",
+                        (void *)m_client);
+                    return nullptr;
```

Comment:
you also need to `aws_raise_error(AWS_ERROR_INVALID_ARGUMENT)`

Reply:
Added!